qualtran/bloqs/block_encoding.py

```diff
@@ -199,18 +264,41 @@ class BlackBoxBlockEncoding(Bloq):
         \mathrm{SELECT} |l\rangle_a|\psi\rangle_s = |l\rangle_a U_l |\psi\rangle_s.
     $$
 
+    The ancilla register is at least of size $\log L$.
+
+    In our implementations we typically split the ancilla registers into
+    selection registers (i.e.  the $l$ registers above) and junk registers which
+    are extra qubits needed by state preparation but not controlled upon during
+    SELECT.
+
+    Here $|G\rangle = \mathrm{PREPARE}|0\rangle$.
+
     Args:
         select: The bloq implementing the `SelectOracle` interface.
-        prepare: The bloq implementing the `SelectOracle` interface.
+        prepare: The bloq implementing the `PrepareOracle` interface.
 
     Registers:
         selection: The combined selection register.
         junk: Additional junk registers not prepared upon.
         system: The combined system register.
+
+    References:
+        [Hamiltonian Simulation by Qubitization](https://quantum-journal.org/papers/q-2019-07-12-163/)
+            Sec 3.1, page 7 and 8 for high level overview and definitions. A
+            block encoding is called a standard form encoding there.
+
+        [The power of block-encoded matrix powers: improved regression techniques via faster Hamiltonian simulation](https://arxiv.org/abs/1804.01973)
+            Definition 3 page 8.
     """
 
-    select: BlackBoxSelect
-    prepare: BlackBoxPrepare
+    alpha: SymbolicFloat
+    epsilon: SymbolicFloat
```

Comment:
The `epsilon` in the LCU Block Encoding can typically be derived from the `epsilon` parameter used for state preparation in the `prepare` oracle. But since `prepare` currently doesn't expose an `epsilon` parameter; we can create an issue to track this and handle it later.

Reply:
It's potentially a combination of epsilons right, if there are multi-stage state preparations. Also I imagine rotation synthesis is also packed into this epsilon?

Reply:
Added #985